### ISSUE TITLE
Red squiggly line in VS when diagnostic is suppressed

### ISSUE THREAD

**Issue by hugener** [dotnet/roslyn#57041]
**Version Used**: 

Microsoft Visual Studio Enterprise 2019 Version 16.11.3
Microsoft.CodeAnalysis Version 3.11.0

**Steps to Reproduce**:

This solution reproduces the issue: https://github.com/dotnet/roslyn/files/7303368/DUTester.zip
It's using TreatWarningsAsErrors, meaning the reported CS8509 would fail the build.

1. At line 14 in Program.cs there is a switch expression which reports a CS8509 warning. The warning is suppressed by another analyzer because it's a discriminated union, where the other analyzer checked that all cases are handled.

**Expected Behavior**:
No squiggly lines on the switch expression

**Actual Behavior**:
Squiggly lines on the switch expression. The build still succeeds, so it appears to be a visualization issue in VS.

**Comment by hugener** [comment]
Any news on this issue?

**Comment by CyrusNajmabadi** [comment]
@hugener No.  This is on the backlog, so it's not something we're actively looking at.
